code/weapon/beam.cpp

```diff
@@ -392,6 +384,21 @@ int beam_fire(beam_fire_info *fire_info)
 
 	for (int i = 0; i < MAX_BEAM_SECTIONS; i++)
 		new_item->beam_section_frame[i] = 0.0f;
+
+	// beam width
+	if (wip->b_info.beam_width > 0.0f) {
+		new_item->beam_width = wip->b_info.beam_width;
+	}
+	else {
+		// lookup
+		float widest = -1.0f;
+		for (int idx = 0; idx < Weapon_info[new_item->weapon_info_index].b_info.beam_num_sections; idx++) {
+			if (Weapon_info[new_item->weapon_info_index].b_info.sections[idx].width > widest) {
+				widest = Weapon_info[new_item->weapon_info_index].b_info.sections[idx].width;
+			}
+		}
+		new_item->beam_width = widest;
+	}
```

Comment:
You may want to keep this in a function even if it's only used once.  It would help organize the code.